### ISSUE TITLE
Send most recent position first

### ISSUE THREAD

**Issue by redge76** [traccar/traccar-client-android#382]
Hi,

If this not an issue on traccar server side, don't you think that in case there is multiple positions waiting in the buffer to be sent, the last know position should be sent first ?
In DatabaseHelper.java add DESC to the line 127.
```sql
SELECT * FROM position ORDER BY id LIMIT 1 DESC
```
(order is only significant if data connection is bad and there is a lot of buffered position but still...)

**Comment by tananaev** [comment]
When data is sent in the wrong order, it results in issues with distance calculation on the server side.

**Comment by redge76** [comment]
:-( ok then close the issue.
I understand it must reduce complexity in server side.

**Comment by oliv3** [comment]
Server doesn't handle out-of-order messages ?

**Comment by tananaev** [comment]
Distance and hours calculation doesn't work correctly if messages are in the wrong order. Everything else works.

**Comment by oliv3** [comment]
That's problematic, and plan to fix this ? Or is there already an issue for it ?

**Comment by tananaev** [comment]
There is a long term plan and there might be an issue for it already somewhere.

**Comment by redge76** [comment]
Well problematic....  Except if a bug in a GPS tracker "decides" to send positions in the wrong order, there is no possibility for out of order messages.
 But yes... If this was handled it would be better.
I use traccar as "safery tracker" in places where data connection is very bad. So I'm very interested to have the latest position each time the client manage to send something.

(as a side note, I tried quite a long time ago to implement a protocol to send positions by SMS (way better for situation where network quality is bad. I forked your project. The idea was to send position by SMS if there was no data network for XX minutes. But it means that some positions will be sent twice )

(Also to have some sort of plugin "api" in your client to be able to extend it without having to create a fork would help but that another  story....)
